File: frontend/webshop/src/app/models/order.ts
import { CartItem } from "./cart-item";
import { Person } from "./person";
import { Review } from "./review";

export class Order {
    id: number;
    createdAt: Date;
    payId: string;
    payerId: string;
    sum: number;
    person: Person;
    items: CartItem[];
    review: Review;
  
    constructor(
      id: number,
      createdAt: Date,
      sum: number,
      person: Person,
      payId: string,
      payerId: string,
      items: CartItem[],
      review: Review,
    ) {
      this.id = id;
      this.createdAt = createdAt;
      this.payId = payId;
      this.payerId = payerId;
      this.items = items;
      this.sum = sum;
      this.person = person;
      this.review = review;
    }
  }
  

File: backend/webshop/order/src/main/java/com/fer/hr/order/model/Order.java
package com.fer.hr.order.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "\"order\"")
public class Order {

    @Id
    @SequenceGenerator(
            name = "order_id_sequence",
            sequenceName = "order_id_sequence"
    )
    @GeneratedValue(
            strategy = GenerationType.IDENTITY,
            generator = "order_id_sequence"
    )
    private Long id;
    private Date createdAt;
    private String payId;
    private String payerId;
    private float sum;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private Person person;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<CartItem> items;
}
